lib/administrate/search.rb

```diff
@@ -44,7 +44,13 @@ def attribute_types
 
     def query_table_name(attr)
       if association_search?(attr)
-        ActiveRecord::Base.connection.quote_table_name(attr.to_s.pluralize)
+        table_name = if attribute_types[attr].options[:class_name].present?
+                       attribute_types[attr].options[:class_name].constantize.table_name
```

Comment:
Since the line `attribute_types[attr].options[:class_name]` is used twice, I think it's worth moving it into a variable.